packages/webview_flutter/webview_flutter_web/lib/webview_flutter_web.dart

```diff
@@ -199,8 +204,11 @@ class WebWebViewPlatformController implements WebViewPlatformController {
     final String contentType =
         httpReq.getResponseHeader('content-type') ?? 'text/html';
     // ignore: unsafe_html
-    _element.src =
-        'data:$contentType,${Uri.encodeFull(httpReq.responseText ?? '')}';
+    _element.src = Uri.dataFromString(
+      httpReq.responseText ?? '',
+      mimeType: contentType,
+      encoding: utf8,
```

Comment:
Quick question. This correctly determines the `mimeType` from the response object. But what if the response is a binary `image/png` instead of `text/html`? Would the Uri encoding work in this case?

(Note: this might have been broken in the earlier version too :S)

Reply:
Worked for me.
I replaced the Url in the example project of webview_flutter_web with one that returns a `content-type` of `image/jpeg` and it displayed the image.